False positives in import/no-unresolved when command is run from path not correctly cased

I was getting a lot of false positives like this:
```
   1:34  error  Casing of react does not match the underlying filesystem                                            import/no-unresolved
   2:25  error  Casing of react-redux does not match the underlying filesystem                                      import/no-unresolved
   3:27  error  Casing of redux-form does not match the underlying filesystem                                       import/no-unresolved
```

In Windows, you have a case-insensitive file system which enables you to enter directories regardless of their casing. I.e. the path: `C:\Git\projects\myproject\src` can be entered like this:

```
cd C:\
cd git <-- this is the culprit
cd projects
cd myproject
cd src
npm run eslint
```

When using your lint rule `import/no-unresolved`, this will crash with the above false positives. They will disappear of you go into the directory `Git` instead of `git`.

I think the lint rule should not go further back than the level where the `package.json` is. This problem will appear when you're using either PowerShell or Bash. And it was a huge pain to figure out...

--- a/tests/src/core/resolve.js
+++ b/tests/src/core/resolve.js
@@ -3,6 +3,7 @@ import { expect } from 'chai'
 import resolve, { CASE_SENSITIVE_FS, fileExistsWithCaseSync } from 'eslint-module-utils/resolve'
 import ModuleCache from 'eslint-module-utils/ModuleCache'
 
+import * as path from 'path'
 import * as fs from 'fs'
 import * as utils from '../utils'
 
@@ -42,6 +43,11 @@ describe('resolve', function () {
     it('detects case does not match FS', function () {
       expect(fileExistsWithCaseSync(file, ModuleCache.getSettings(testContext)))
         .to.be.false
+    })
+    it('detecting case does not include parent folder path (issue #720)', function () {
+      const f = path.join(process.cwd().toUpperCase(), './tests/files/jsx/MyUnCoolComponent.jsx')
+      expect(fileExistsWithCaseSync(f, ModuleCache.getSettings(testContext), true))
+        .to.be.true
     })
   })
 

--- a/utils/resolve.js
+++ b/utils/resolve.js
@@ -21,6 +21,7 @@ exports.fileExistsWithCaseSync = function fileExistsWithCaseSync(filepath, cache
 
   // null means it resolved to a builtin
   if (filepath === null) return true
+  if (filepath.toLowerCase() === process.cwd().toLowerCase()) return true
   const parsedPath = path.parse(filepath)
       , dir = parsedPath.dir
 

